### PR TITLE
qmllive: add new recipe

### PR DESCRIPTION
**Pull request by sukdo399**
qmlLive is a live reloader environment for rapid UI development

Change-Id: Icc29a45ed488a95257625122107b83cbe746c755
Signed-off-by: Seokha Ko <sukdo399@daum.net>

### CODE REVIEW

File: recipes-qt/qmllive/qmllive_git.bb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+DESCRIPTION = "QmlLive is a live reloader environment for rapid UI development"
+HOMEPAGE = "https://github.com/qtproject/qt-apps-qmllive"
+LICENSE = "GPL-3.0"
+SRC_URI = "git://code.qt.io/qt-apps/qmllive.git;branch=dev"
+
+LIC_FILES_CHKSUM = " \
+    file://LICENSE.GPL3;md5=75cd0dbc6f2d24e7eeb128ff59b74f4c \
+"
```
Comment (otavio) [inline recipes-qt/qmllive/qmllive_git.bb]: `LIC_FILES_CHKSUM` must go next to `LICENSE`.

File: recipes-qt/qmllive/qmllive_git.bb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+DEPENDS = "qtbase qtdeclarative"
```
Comment (otavio) [inline recipes-qt/qmllive/qmllive_git.bb]: `DEPENDS` after license block.

File: recipes-qt/qmllive/qmllive_git.bb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+DESCRIPTION = "QmlLive is a live reloader environment for rapid UI development"
+HOMEPAGE = "https://github.com/qtproject/qt-apps-qmllive"
+LICENSE = "GPL-3.0"
+SRC_URI = "git://code.qt.io/qt-apps/qmllive.git;branch=dev"
```
Comment (otavio) [inline recipes-qt/qmllive/qmllive_git.bb]: `SRC_URI` before `SRCREV`

File: recipes-qt/qmllive/qmllive_git.bb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    file://LICENSE.GPL3;md5=75cd0dbc6f2d24e7eeb128ff59b74f4c \
+"
+DEPENDS = "qtbase qtdeclarative"
+
+S = "${WORKDIR}/git"
+SRCREV = "0c7bf141b08aa9e757e91a4a05769257d043eab2"
+PV = "git${SRCPV}"
+
+inherit pkgconfig
+inherit qmake5
```
Comment (otavio) [inline recipes-qt/qmllive/qmllive_git.bb]: Put the classes on single line.

Comment (sukdo399) [reply recipes-qt/qmllive/qmllive_git.bb]: It was fixed.

File: recipes-qt/qmllive/qmllive_git.bb (hunk 3 above)
Comment (sukdo399) [reply recipes-qt/qmllive/qmllive_git.bb]: It was fixed.

File: recipes-qt/qmllive/qmllive_git.bb (hunk 2 above)
Comment (sukdo399) [reply recipes-qt/qmllive/qmllive_git.bb]: It was fixed.

File: recipes-qt/qmllive/qmllive_git.bb (hunk 1 above)
Comment (sukdo399) [reply recipes-qt/qmllive/qmllive_git.bb]: It was fixed.

**Comment by sukdo399** [comment]
@otavio 
Thank you for your review.
Style issues were fixed based on guidelines.

File: recipes-qt/qmllive/qmllive_git.bb [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+LICENSE = "GPL-3.0"
+LIC_FILES_CHKSUM = " \
+    file://LICENSE.GPL3;md5=75cd0dbc6f2d24e7eeb128ff59b74f4c \
+"
+
+DEPENDS = "qtbase qtdeclarative"
+
+PV = "git${SRCPV}"
+
+SRC_URI = "git://code.qt.io/qt-apps/qmllive.git;branch=dev"
```
Comment (shr-project) [inline recipes-qt/qmllive/qmllive_git.bb]: You can use QT_GIT variable here.

File: recipes-qt/qmllive/qmllive_git.bb [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+SRCREV = "0c7bf141b08aa9e757e91a4a05769257d043eab2"
+S = "${WORKDIR}/git"
+
+inherit pkgconfig qmake5
+
+EXTRA_QMAKEVARS_POST = "-after QMAKE_RPATHDIR="
+
+FILES_${PN}-dev += "/usr/share/*"
```
Comment (shr-project) [inline recipes-qt/qmllive/qmllive_git.bb]: Doesn't it respect datadir? And the * isn't necessary.

File: recipes-qt/qmllive/qmllive_git.bb [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+HOMEPAGE = "https://github.com/qtproject/qt-apps-qmllive"
+
+LICENSE = "GPL-3.0"
+LIC_FILES_CHKSUM = " \
+    file://LICENSE.GPL3;md5=75cd0dbc6f2d24e7eeb128ff59b74f4c \
+"
+
+DEPENDS = "qtbase qtdeclarative"
+
+PV = "git${SRCPV}"
```
Comment (shr-project) [inline recipes-qt/qmllive/qmllive_git.bb]: Doesn't it have some more meaningful version?

File: recipes-qt/qmllive/qmllive_git.bb (hunk 5 above)
Comment (sukdo399) [reply recipes-qt/qmllive/qmllive_git.bb]: fixed.

File: recipes-qt/qmllive/qmllive_git.bb (hunk 6 above)
Comment (sukdo399) [reply recipes-qt/qmllive/qmllive_git.bb]: fixed.

File: recipes-qt/qmllive/qmllive_git.bb (hunk 7 above)
Comment (sukdo399) [reply recipes-qt/qmllive/qmllive_git.bb]: fixed.

File: recipes-qt/qmllive/qmllive_git.bb [hunk 8]
```diff
@@ -0,0 +1,21 @@
+DESCRIPTION = "QmlLive is a live reloader environment for rapid UI development"
+HOMEPAGE = "https://github.com/qtproject/qt-apps-qmllive"
+
```
[the remark this answers is not in the document]
Comment (sukdo399) [reply recipes-qt/qmllive/qmllive_git.bb]: fixed.

File: recipes-qt/qmllive/qmllive_git.bb [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+LICENSE = "GPL-3.0"
+LIC_FILES_CHKSUM = " \
+    file://LICENSE.GPL3;md5=75cd0dbc6f2d24e7eeb128ff59b74f4c \
+"
```
[the remark this answers is not in the document]
Comment (sukdo399) [reply recipes-qt/qmllive/qmllive_git.bb]: fixed.

**Comment by sukdo399** [comment]
@otavio @shr-project 
Thank you for the review.
Issues were fixed so please check again.

File: recipes-qt/qmllive/qmllive_git.bb [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+PV = "5.12+git${SRCPV}"
+
+QT_GIT_PROJECT = "qt-apps"
+SRC_URI = "${QT_GIT}/qmllive.git;branch=dev"
+SRCREV = "0c7bf141b08aa9e757e91a4a05769257d043eab2"
+S = "${WORKDIR}/git"
+
+inherit pkgconfig qmake5
+
+EXTRA_QMAKEVARS_POST = "-after QMAKE_RPATHDIR="
```
Comment (shr-project) [inline recipes-qt/qmllive/qmllive_git.bb]: Isn't -after already there from AFTER variable?

**Comment by sukdo399** [comment]
@shr-project 
Could you please review again?

**Comment by otavio** [comment]
All issues were addressed. Thank you @sukdo399 for this patch and for being responsive with the needed rework points.

**Pull request merged**